infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -323,6 +317,22 @@ public void handleTokenQuery(CorfuPayloadMsg<TokenRequest> msg,
                 TokenType.NORMAL, TokenResponse.NO_CONFLICT_KEY, token, Collections.emptyMap())));
     }
 
+    @ServerHandler(type = CorfuMsgType.TRIM_SEQUENCER_REQ, opTimer = metricsPrefix + "trimCache")
+    public synchronized void trimCache(CorfuPayloadMsg<Long> msg,
+                                       ChannelHandlerContext ctx, IServerRouter r,
+                                       boolean isMetricsEnabled) {
+        long trimMark = msg.getPayload();
+        long entries = 0;
+        for (Map.Entry<String, Long> entry : conflictToGlobalTailCache.asMap().entrySet()) {
```

Comment:
you want to log some metrics here because this might disrupt the sequencer performance at times. Add per invalidate timing also. If it is slow we might have to do this in smaller batches or something to that effect.